driver-core/src/test/functional/com/mongodb/ClusterFixture.java

```diff
@@ -349,7 +349,7 @@ public static StreamFactory getStreamFactory() {
     }
 
     public static StreamFactory getAsyncStreamFactory() {
-        String streamType = System.getProperty("org.mongodb.async.type", "nio2");
+        String streamType = System.getProperty("org.mongodb.test.async.type", "nio2");
```

Comment:
I think you may need to set the property in the top level:  build.gradle

Reply:
Yep, I neglected to change the name there.  Fixed in latest commit, and tested manually from command line to make sure the system property was getting set properly.